nautobot/extras/models/groups.py

```diff
@@ -17,16 +18,140 @@
+        """Return the member objects for this group."""
+        # Since associated_object is a GenericForeignKey, we can't just do:
+        #     return self.static_group_associations.values_list("associated_object", flat=True)
+        return self.model.objects.filter(
+            pk__in=self.static_group_associations.values_list("associated_object_id", flat=True)
+        )
+
+    def clean(self):
+        super().clean()
+
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+
+            if self.content_type != database_object.content_type:
+                raise ValidationError({"content_type": "ContentType cannot be changed once created"})
+
+    def _check_valid_member(self, member):
+        if not isinstance(member, self.model):
+            raise TypeError(f"{member} is not a {self.model._meta.label_lower}")
+
+    def add_members(self, objects_to_add):
+        """Add the given list or QuerySet of objects to this Static Group."""
+        for obj in objects_to_add:
+            self._check_valid_member(obj)
+
+        for obj in objects_to_add:
+            # We don't use `.bulk_create()` currently because we want change logging for these creates.
+            # Might be a good future performance improvement though.
+            StaticGroupAssociation.objects.get_or_create(
+                static_group=self, associated_object_type=self.content_type, associated_object_id=obj.pk
+            )
+
+    def remove_members(self, objects_to_remove):
+        """Remove the given list or QuerySet of objects from this Static Group."""
+        pks_to_remove = set()
+        for obj in objects_to_remove:
+            self._check_valid_member(obj)
+            pks_to_remove.add(obj.pk)
+
+        StaticGroupAssociation.objects.filter(
+            static_group=self, associated_object_type=self.content_type, associated_object_id__in=pks_to_remove
+        ).delete()
+
+
+@extras_features(
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class StaticGroupAssociation(OrganizationalModel):
+    """Intermediary model for associating an object to a StaticGroup."""
+
+    static_group = models.ForeignKey(to=StaticGroup, on_delete=models.CASCADE, related_name="static_group_associations")
+    associated_object_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        related_name="static_group_associations",
+        limit_choices_to=StaticGroupModelsQuery(),
+    )
+    associated_object_id = models.UUIDField(db_index=True)
+    associated_object = GenericForeignKey(ct_field="associated_object_type", fk_field="associated_object_id")
+
+    is_static_group_associable_model = False
+
+    class Meta:
+        unique_together = ("static_group", "associated_object_type", "associated_object_id")
+
+    def __str__(self):
+        return f"{self.associated_object} ∈ {self.static_group}"
```

Comment:
What's this `∈` symbol?

Reply:
Set theory "is a member of" or "belongs to". As I mentioned in dojo yesterday I was expecting this might be too academic/nerdy in general. Do you think `f"{self.associated_object} as a member of {self.static_group}"` would be better?

Reply:
Oh, right! Might be better to avoid Unicode in general and yeah it wasn't clear to me, a humble network engineer